### PR TITLE
Declare iterators using auto

### PR DESCRIPTION
**Pull request by negatratoron**
Just some janitorial work.  Also made some slight consistency adjustments such as in crypto_test.cpp.

Edit: apparently can't edit files in the leveldb subtree, also my test system didn't build zeromq so i didn't catch my syntax error.

### CODE REVIEW

**Comment by promag** [comment]
NACK, I think this makes code less readable. Usually we argue about `auto` usage in new code, but I don't recall of changing code to use `auto`.

My preference is to use `auto` like `auto x = new Foo` so it's clear the `x` type.

File: src/addrman.cpp [hunk 1]
```diff
@@ -67,12 +67,12 @@ double CAddrInfo::GetChance(int64_t nNow) const
 
 CAddrInfo* CAddrMan::Find(const CNetAddr& addr, int* pnId)
 {
-    std::map<CNetAddr, int>::iterator it = mapAddr.find(addr);
+    auto it = mapAddr.find(addr);
```
Comment (promag) [inline src/addrman.cpp]: I'd rater see this changed to `std::map<CNetAddr, int>::const_iterator it = ...` for instance.

**Comment by jamesob** [comment]
NACK. 

From the pull request template:
> Refactoring changes are only accepted if they are required for a feature or bug fix or otherwise improve developer experience significantly. For example, most "code style" refactoring changes require a thorough explanation why they are useful, what downsides they have and why they significantly improve developer experience or avoid serious programming bugs. Note that code style is often a subjective matter. Unless they are explicitly mentioned to be preferred in the developer notes, stylistic code changes are usually rejected.

File: src/addrman.cpp (hunk 1 above)
Comment (practicalswift) [reply src/addrman.cpp]: The following `scripted-diff` compiles if someone wants to look for `::const_iterator` candidates:

```
git grep -lE '::iterator ' "*.cpp" "*.h" | \
    grep -vE '(crypto_tests.cpp|coins_tests.cpp|limitedmap.h|guiutil.cpp|addrman.cpp|bloom.cpp|net.cpp|net_processing.cpp|coins.cpp|coins.h)' | \
    xargs sed -i 's/::iterator /::const_iterator /g'
```

**Comment by jeffersoncarpenter** [comment]
CI builds for this patch took 2X-3X longer than builds for contemporary patches, and one of them failed due to timeout (I don't know how that CI build succeeded thereafter).

**Comment by promag** [comment]
@jeffersoncarpenter CI time varies and depends on current build cache. Sometimes failure builds are restarted manually by people that has the right permission.

However, after the above comments, I think you should close this and follow @jamesob advice or @practicalswift https://github.com/bitcoin/bitcoin/pull/15762#discussion_r273196897 suggestion.

**Comment by jeffersoncarpenter** [comment]
Closing.

**Pull request closed without merging**